File: copilot-bot/src/commands/help.ts
import type { CommandHandler } from "@woife5/shared";
import { type ChatInputCommandInteraction, EmbedBuilder, SlashCommandBuilder } from "discord.js";
import { COPILOT_ICON, MESSAGE } from "../consants";

export const help: CommandHandler = {
    data: new SlashCommandBuilder().setName("help").setDescription("Get help with Microsoft Copilot."),
    executeInteraction: async (interaction: ChatInputCommandInteraction): Promise<void> => {
        await interaction.reply({ embeds: [runCommand()] });
    },
};

function runCommand() {
    return new EmbedBuilder()
        .setColor("White")
        .setAuthor({ name: "Copilot", iconURL: COPILOT_ICON })
        .setTitle("Microsoft Copilot")
        .setDescription(MESSAGE);
}

File: copilot-bot/commands/help.go
package commands

import (
	"github.com/bwmarrin/discordgo"
	"github.com/woife5/copilot-bot/config"
)

func (b *Bot) handleHelp(s *discordgo.Session, i *discordgo.InteractionCreate) {
	embed := &discordgo.MessageEmbed{
		Color: 0xFFFFFF, // White
		Author: &discordgo.MessageEmbedAuthor{
			Name:    "Copilot",
			IconURL: config.CopilotIcon,
		},
		Title:       "Microsoft Copilot",
		Description: config.FallbackMessage,
	}

	s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
		Type: discordgo.InteractionResponseChannelMessageWithSource,
		Data: &discordgo.InteractionResponseData{
			Embeds: []*discordgo.MessageEmbed{embed},
		},
	})
}
